### PR TITLE
Homework 4

### PR DESCRIPTION
**Pull request by Tailor-AI**
Added the requirements to RecipeApp solution from lesson 4.

### CODE REVIEW

File: RecipeApp.Client/Form1.cs [hunk 1]
```diff
@@ -25,30 +20,44 @@ private void LoadButton_Click(object sender, EventArgs e)
...
+                        using (var reader = new StreamReader(stream))
+                        {
+                            return reader.ReadToEnd();
+                        }
+                    }
+                    else
+                    {
+                        MessageBox.Show("Invalid recipe file. Must end with .recipe");
                     }
                 }
```
Comment (Almantask) [inline RecipeApp.Client/Form1.cs]: Dialog has an option to filter on initial path and extensions, try to make use of that 😄

File: RecipeApp.Core/RecipeConverter.cs [hunk 2]
```diff
@@ -76,16 +57,43 @@ static void StandardiseCookingUnit(int index, string[] words)
...
+        {
+            if (amountMl >= 100)
+            {
+                return (amountMl / 1000).ToString("F2", CultureInfo.InvariantCulture);
+            }
+            else
+            {
+                return amountMl.ToString("F2", CultureInfo.InvariantCulture);
+            }
+        }
```
Comment (Almantask) [inline RecipeApp.Core/RecipeConverter.cs]: can be simplified:
```cs
        private static string GetAmountInSiUnits(double amountMl)
        {
            if(amountMl >= 100)
            {
                amountMl = amountMl / 100;
            }
            return amountMl.ToString("F2", CultureInfo.InvariantCulture);
        }
```

File: RecipeApp.Core/RecipeConverter.cs [hunk 3]
```diff
@@ -94,29 +102,59 @@ static double FindMultiplier(string cookingUnit)
...
+
+            if (!containsCookingOrSiUnits)
+            {
+                throw new InvalidRecipeException("No SI units in recipe.");
+            }
+
+            return string.Join(" ", words);
+        }
+
+        private static void StandardiseSiUnit(int index, string[] words)
```
Comment (Almantask) [inline RecipeApp.Core/RecipeConverter.cs]: Instead, make this method return  a bool so you won't need to use a field but rather have it controlled in local scope. That confused me a bit.

**Review by Almantask — CHANGES_REQUESTED** [review]
Try to slightly be more clear with error messages and please use dialog-native validation.

File: RecipeApp.Client/Form1.cs (hunk 1 above)
Comment (Tailor-AI) [reply RecipeApp.Client/Form1.cs]: Implemented and removed the previous check.

File: RecipeApp.Core/RecipeConverter.cs (hunk 2 above)
Comment (Tailor-AI) [reply RecipeApp.Core/RecipeConverter.cs]: Ahh. Didnt think of that. Thx!

File: RecipeApp.Core/RecipeConverter.cs (hunk 3 above)
Comment (Tailor-AI) [reply RecipeApp.Core/RecipeConverter.cs]: Have removed the field, made function return bool and moved the checking of different units into respective function.

**Comment by Almantask** [comment]
Looks good now :)

**Pull request closed without merging**